docs/reference/query-dsl/term-query.asciidoc

```diff
@@ -1,168 +1,59 @@
-**************************************************
-
-String fields can be of type `text` (treated as full text, like the body of an
-email), or `keyword` (treated as exact values, like an email address or a
-zip code).  Exact values (like numbers, dates, and keywords) have
-the exact value specified in the field added to the inverted index in order
-to make them searchable.
+You can use the `term` query to find documents based on a precise value such as
+a price, a product ID, or a username.
 
-However, `text` fields are `analyzed`. This means that their
-values are first passed through an <<analysis,analyzer>> to produce a list of
-terms, which are then added to the inverted index.
+[WARNING]
+====
+Avoid using the `term` query for <<text, `text`>> fields.
 
-There are many ways to analyze text: the default
-<<analysis-standard-analyzer,`standard` analyzer>> drops most punctuation,
-breaks up text into individual words, and lower cases them.    For instance,
-the `standard` analyzer would turn the string ``Quick Brown Fox!'' into the
-terms [`quick`, `brown`, `fox`].
+By default, {es} changes the values of `text` fields as part of <<analysis,
+analysis>>. This can make finding exact matches for `text` field values
+difficult.
 
-This analysis process makes it possible to search for individual words
-within a big block of full text.
+To search `text` field values, use the <<query-dsl-match-query,`match`>> query
+instead.
+====
 
-The `term` query looks for the *exact* term in the field's inverted index --
-it doesn't know anything about the field's analyzer.  This makes it useful for
-looking up values in keyword fields, or in numeric or date
-fields.  When querying full text fields, use the
-<<query-dsl-match-query,`match` query>> instead, which understands how the field
-has been analyzed.
-
-
-To demonstrate, try out the example below.  First, create an index, specifying the field mappings, and index a document:
+==== Example request
 
 [source,js]
---------------------------------------------------
-PUT my_index
-{
-  "mappings": {
-    "properties": {
-      "full_text": {
-        "type":  "text" <1>
-      },
-      "exact_value": {
-        "type":  "keyword" <2>
-      }
+----
+GET /_search
+{
+    "query": {
+        "term": {
+            "user": {
+                "value": "Kimchy",
+                "boost": 1.0
+            }
+        }
     }
-  }
 }
-
-PUT my_index/_doc/1
-{
-  "full_text":   "Quick Foxes!", <3>
-  "exact_value": "Quick Foxes!"  <4>
-}
---------------------------------------------------
+----
 // CONSOLE
 
-<1> The `full_text` field is of type `text` and will be analyzed.
-<2> The `exact_value` field is of type `keyword` and will NOT be analyzed.
-<3> The `full_text` inverted index will contain the terms: [`quick`, `foxes`].
-<4> The `exact_value` inverted index will contain the exact term: [`Quick Foxes!`].
-
-Now, compare the results for the `term` query and the `match` query:
```

Comment:
Just out of curiosity: are there follow up plans to move these more detailed examples somewhere else? While I like the idea of having succinct, standardized docs for each query, I find these kind of examples quite useful and better to understand than merely the minimal snippet that remains. Not saying this shouldn't go away, I'm just curious what the plan is for examples like this in this rewriting effort.

Reply:
Thanks for the feedback @cbuescher.

For this particular set of examples, I think a concept-focused page like "Search structured data" or "Search full-text" would be a better fit. Those pages don't exist yet, but I'll work on creating them and add it to this PR.

For other queries, I think we can include additional sections for detailed examples below parameter documentation. A good example of this would be the [`missing` query section](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-exists-query.html#_literal_missing_literal_query) for the `exists` query.

Let me know if you feel differently. I'm still new so there could be context I'm missing.